### ISSUE TITLE
TypeError when Algolia data containing enum field

### ISSUE THREAD

**Issue by ryutosh** [FlutterFlow/flutterflow-issues#3638]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

If the data returned from Algolia contains an Enum field, the results cannot be deserialized as a document, even though Algolia has returned search results.

As a result, search results cannot be displayed in Google Maps, etc.
(It is determined as error, and the map is not displayed and continues in circular progress).

This is a common problem for web and real devices.

The type masked in the message below is Enum. The error message is saying that it is not a subset of String.

![image](https://github.com/user-attachments/assets/64861520-65a5-4aaf-b7c9-9543604bee48)

### Expected Behavior

Desirialized as Enum without error.

### Steps to Reproduce

1. Create a Firestore Collection using Enum
2. Create Algolia Index for the collection
3. Create an widget of Google Maps and an action for Algolia search
4. Set map marker from algolia results

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITEehfKB35ZJpbwG1c7qcsJWiQMwGm45a+cZkdV8EBAaCePnPLMQdcj8QElvWOqHTwlmGmKGgTkG3M71ktrLFewoYAmefL5FwbVbUxDiRXGlVqrUCpDtQXB7J8ZVGnaOy5iVqx1NANBpc2Qi2k6MHu+Qb3qCf9qOYwx5e6fDbOY=

### Visual documentation

In current behavior section

### Environment

```markdown
- FlutterFlow version: 4.1.82
- Platform: MacOS
- Browser name and version: Chrome
- Operating system and version affected: Mac Sonoma
```

### Additional Information

I assume that FF may have made a mistake in handling the following Issue, then causing another problem.
https://github.com/FlutterFlow/flutterflow-issues/issues/3464

**Comment by Alezanello** [comment]
Hello!

This issue seems very similar to [issue #3464](https://github.com/FlutterFlow/flutterflow-issues/issues/3464) regarding how Algolia is serializing the Enum in Firestore. I’ll keep you updated with any new information in that thread and will close this one as a duplicate.

**Comment by ryutosh** [comment]
@Alezanello 
OK, please be sure that the cause of this issue also fixes in issue #3464 .
